### PR TITLE
Tgui-next pandemic

### PR DESCRIPTION
**Pull request by actioninja**
## About The Pull Request

image

Also a refactor to how thresholds are stored and sent to allow better flexibility on end display

## Changelog
:cl:
add: pandemic now uses tgui-next
/:cl:

### CODE REVIEW

File: tgui-next/packages/tgui/interfaces/Pandemic.js [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+            <LabeledList.Item label="Stage Speed">
+              {stage_speed}
+            </LabeledList.Item>
+            <LabeledList.Item label="Transmission">
+              {transmission}
+            </LabeledList.Item>
+          </LabeledList>
+        </Grid.Column>
+      </Grid>
+      {thresholds !== {} && (
```
Comment (stylemistake) [inline tgui-next/packages/tgui/interfaces/Pandemic.js]: You can't compare objects like this. Equality compares object identities (pointers), not contents. One way you could do it cleanly, is to convert it to a list first (using the same `map()`, but near your data destructurings), then check `length` property on it.

Something like:
```js
const thresholds = map((desc, label) => ({ desc, label }))(data.thresholds || {});
```

As a bonus, you get a clean array to work with. But ideally, of course, you want backend to send clean arrays:

```dm
threshold_descs = list(
  list(
    "label" = "Resistance 8"
    "desc" = "Additionally regenerates lost blood."
  ),
  ...
)
```

**Review by stylemistake — CHANGES_REQUESTED** [review]

**Comment by AnturK** [comment]
Needs update

**Comment by Shirbu** [comment]
You can no longer name diseases.

**Comment by stylemistake** [comment]
@actioninja What happened to the input. I merged it before realising it wasn't there.

**Comment by actioninja** [comment]
It should be being dynamically replaced with the input if it's renamable, then replaced with normal text if you can't rename it. iirc you weren't able to rename the stock flu before as well.

**Pull request merged**